bluetooth/src/androidLibMain/kotlin/scanner/Scanner.kt

```diff
@@ -0,0 +1,191 @@
+import com.splendo.kaluga.bluetooth.device.DeviceConnectionManager
+import com.splendo.kaluga.bluetooth.device.DeviceHolder
+import com.splendo.kaluga.bluetooth.device.DeviceInfoImpl
+import com.splendo.kaluga.location.LocationEnabledMonitor
+import com.splendo.kaluga.permissions.Permission
+import com.splendo.kaluga.permissions.Permissions
+import com.splendo.kaluga.state.StateRepo
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import no.nordicsemi.android.support.v18.scanner.BluetoothLeScannerCompat
+import no.nordicsemi.android.support.v18.scanner.ScanCallback
+import no.nordicsemi.android.support.v18.scanner.ScanFilter
+import no.nordicsemi.android.support.v18.scanner.ScanResult
+import no.nordicsemi.android.support.v18.scanner.ScanSettings
+
+actual class Scanner internal constructor(
+    private val bluetoothScanner: BluetoothLeScannerCompat = BluetoothLeScannerCompat.getScanner(),
+    private val bluetoothAdapter: BluetoothAdapter? = BluetoothAdapter.getDefaultAdapter(),
+    private val scanSettings: ScanSettings = defaultScanSettings,
+    private val context: Context = ApplicationHolder.applicationContext,
+    permissions: Permissions,
+    connectionSettings: ConnectionSettings,
+    autoRequestPermission: Boolean,
+    autoEnableBluetooth: Boolean,
+    stateRepo: StateRepo<ScanningState>,
+    private val coroutineScope: CoroutineScope
+) : BaseScanner(permissions, connectionSettings, autoRequestPermission, autoEnableBluetooth, stateRepo, coroutineScope) {
+
+    class Builder(
+        private val bluetoothScanner: BluetoothLeScannerCompat = BluetoothLeScannerCompat.getScanner(),
+        private val bluetoothAdapter: BluetoothAdapter? = BluetoothAdapter.getDefaultAdapter(),
+        private val scanSettings: ScanSettings = defaultScanSettings,
+        private val context: Context = ApplicationHolder.applicationContext
+    ) : BaseScanner.Builder {
+
+        override fun create(
+            permissions: Permissions,
+            connectionSettings: ConnectionSettings,
+            autoRequestPermission: Boolean,
+            autoEnableBluetooth: Boolean,
+            scanningStateRepo: StateRepo<ScanningState>,
+            coroutineScope: CoroutineScope
+        ): BaseScanner {
+            return Scanner(bluetoothScanner, bluetoothAdapter, scanSettings, context, permissions, connectionSettings, autoRequestPermission, autoEnableBluetooth, scanningStateRepo, coroutineScope)
+        }
+    }
+
+    companion object {
+        val defaultScanSettings = ScanSettings.Builder()
+            .setScanMode(ScanSettings.SCAN_MODE_BALANCED)
+            .setNumOfMatches(ScanSettings.MATCH_NUM_FEW_ADVERTISEMENT)
+            .build()
+    }
+
+    private val callback = object : ScanCallback() {
+
+        override fun onScanFailed(errorCode: Int) {
+            val error = when (errorCode) {
+                SCAN_FAILED_ALREADY_STARTED -> Pair("Already Started", false)
```

Comment:
You have guess what is second parameter is. Some comment or data class will help